homeassistant/components/overkiz/config_flow.py

```diff
@@ -1,78 +1,221 @@
+                except NotSuchTokenException:
+                    LOGGER.debug("Token is invalid / has been already removed")
+
+                raise exception
 
-        await client.login(register_event_listener=False)
+            user_input[CONF_TOKEN] = token
+            user_input[CONF_TOKEN_UUID] = uuid
 
-        # Set first gateway id as unique id
+        else:
+            server = SUPPORTED_SERVERS[user_input[CONF_SERVER]]
+
+            session = async_create_clientsession(self.hass)
+            client = OverkizClient(
+                username=username, password=password, server=server, session=session
+            )
+
+            await client.login(register_event_listener=False)
+
+        # Set main gateway id as unique id
         if gateways := await client.get_gateways():
-            gateway_id = gateways[0].id
-            await self.async_set_unique_id(gateway_id)
+            for gateway in gateways:
+                if is_overkiz_gateway(gateway.id):
+                    gateway_id = gateway.id
+                    await self.async_set_unique_id(gateway_id)
+
+        return user_input
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step via config flow."""
-        errors = {}
+        errors: dict[str, str] = {}
+
+        if user_input:
+            self._server = user_input[CONF_SERVER]
+
+            # Some Overkiz hubs do support a local API
+            # Users can choose between local or cloud API.
+            if self._server in SERVERS_WITH_LOCAL_API:
+                return await self.async_step_local_or_cloud()
+
+            return await self.async_step_cloud()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_SERVER, default=self._server): vol.In(
+                        {key: hub.name for key, hub in SUPPORTED_SERVERS.items()}
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_local_or_cloud(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Users can choose between local API or cloud API via config flow."""
+        if user_input:
+            self._api_type = user_input[CONF_API_TYPE]
+
+            if self._api_type == APIType.LOCAL:
+                return await self.async_step_local()
+
+            return await self.async_step_cloud()
+
+        return self.async_show_form(
+            step_id="local_or_cloud",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_TYPE): vol.In(
+                        dict(
+                            {
+                                APIType.LOCAL: "Local API",
+                                APIType.CLOUD: "Cloud API",
+                            }.items()
+                        )
```

Comment:
Is it just me or did this already start as a dict, being transformed to items and then casted to a dict?

Reply:
Oops 🙄. Fixed in 89981d36b78fc1a8049bee7e6eff4a5926a700a7.